### DISCUSSION TITLE
Disable Code Analysis for External Headers

### DISCUSSION THREAD

**Discussion by terids** [premake/premake-core#1996] [Q&A]
In VS2022 it is now possible to set this setting under C/C++ -> External Includes. 
I can't find anywhere to set this in Premake5. Is it possible?

I have set `externalwarnings "off"` and added the includes to `externalincludedirs` but can't find a setting for disabling code analysis, or a way to set it manually.

Thanks :)

**Comment by Jarod42** [comment]
Equivalent command line build option is `/analyze:external-`, so you might use:

```lua
filter "toolset:msc*"
  buildoptions "/analyze:external-"
```

Notice that the option won't be visible under `C/C++ -> External Includes` but under `C/C++ -> Command Line`.

As a note, option [exists since VS2019 16.10](https://learn.microsoft.com/en-us/cpp/build/reference/analyze-code-analysis?view=msvc-160#external-file-analysis-options).
